proposals/wg-reference-spec.md

```diff
@@ -0,0 +1,82 @@
+# OCI Working Group: Reference Spec
+
+## Reference Spec OCI Working Group - Governance Charter
+
+This document describes the basic governance principles for the Reference Spec Working Group (the “WG”).
+
+The WG operates as an OCI Working Group under the [Open Container Initiative (OCI) Charter](https://github.com/opencontainers/tob/blob/master/CHARTER.md), which describes the responsibilities of the OCI Technical Oversight Board (the "TOB”). The WG is established by the TOB as an OCI Working Group pursuant to the OCI Charter. Accordingly, the WG will operate in accordance with the OCI Charter and OCI's other policies and procedures, supplemented by the details below.
+
+## Purpose
+
+References are a string that is used by runtimes and other OCI registry clients to retrieve a container image.
+They are currently a convention, used by many clients, adopted from Docker and implemented in distribution/distribution.
+This WG seeks to define the syntax and parsing of a reference as an OCI spec.
+
+## Scope
+
+* Document the existing convention used by runtimes and other OCI registry clients.
+* Specify the syntax for references to an [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/7b36cea86235157d78528944cb94c3323ee0905c/image-layout.md) manifest, including support for a tag, digest, or a full registry/repository name.
+* Define how the syntax can be extended to support other use cases, including:
+  * Alternate layer formats
+  * Immutable tags
+  * Selecting an artifact that refers to another manifest (e.g. signature or SBOM)
+  * Querying content from a pull through cache serving multiple registries (see [distribution-spec PR #66](https://github.com/opencontainers/distribution-spec/pull/66))
+* Provide backwards compatibility by using the existing reference convention from [distribution/reference](https://github.com/distribution/reference) when feasible.
+
+## Out of Scope
+
+* This WG will be limited to the creation of a new spec, and is not expected to impact other OCI specs.
```

Comment:
As noted above, I strongly disagree with this conclusion, but also believe it's premature to decide that this should be a completely net-new spec.

> The goal here is essentially to take the Grammar from https://pkg.go.dev/github.com/distribution/reference@v0.6.0#pkg-overview and write slightly more words around it, especially describing the docker.io, docker.io/library/, and :latest defaults, right?

This isn't very big, and I don't think we should encourage it to be.

Reply:
I don't think the size of the work should determine whether it's a new spec, but rather whether it fits within the definition of an existing spec. Looking at image-spec:

> The OCI Image Format project creates and maintains the software shipping container image format spec (OCI Image Format).

I don't think we fit there. And for distribution-spec:

> The OCI Distribution Spec project defines an API protocol to facilitate and standardize the distribution of content.

We only fit there if we are deciding in advance to only allow references to content on a registry, and not to any other way of storing images, e.g. the OCI Image Layout.